public/app/features/alerting/unified/components/rules/MultipleDataSourcePicker.tsx

```diff
@@ -0,0 +1,165 @@
+
+  const onChange = (items: Array<SelectableValue<string>>, actionMeta: ActionMeta) => {
+    if (actionMeta.action === 'clear' && props.onClear) {
+      props.onClear();
+      return;
+    }
+
+    const selectedItem = items[items.length - 1];
+
+    let dataSourceName, action: 'add' | 'remove';
+
+    if (actionMeta.action === 'pop-value' || actionMeta.action === 'remove-value') {
+      const castedActionMeta:
+        | RemoveValueActionMeta<SelectableValue<string>>
+        | PopValueActionMeta<SelectableValue<string>> = actionMeta;
+      dataSourceName = castedActionMeta.removedValue?.value;
+      action = 'remove';
+    } else {
+      dataSourceName = selectedItem.value;
+      action = 'add';
+    }
+
+    const dsSettings = dataSourceSrv.getInstanceSettings(dataSourceName);
+
+    if (dsSettings) {
+      props.onChange(dsSettings, action);
+      setState({ error: undefined });
+    }
+  };
+
+  const getCurrentValue = (): Array<SelectableValue<string>> | undefined => {
+    const { current, hideTextValue, noDefault } = props;
+    if (!current && noDefault) {
+      return;
+    }
+
+    return current?.map((dataSourceName: string) => {
+      const ds = dataSourceSrv.getInstanceSettings(dataSourceName);
+      if (ds) {
+        return {
+          label: ds.name.slice(0, 37),
+          value: ds.name,
+          imgUrl: ds.meta.info.logos.small,
+          hideText: hideTextValue,
+          meta: ds.meta,
+        };
+      }
+
+      const uid = getDataSourceUID(dataSourceName);
+
+      if (uid === ExpressionDatasourceRef.uid || uid === ExpressionDatasourceRef.name) {
+        return { label: uid, value: uid, hideText: hideTextValue };
+      }
+
+      return {
+        label: (uid ?? 'no name') + ' - not found',
+        value: uid ?? undefined,
+        imgUrl: '',
+        hideText: hideTextValue,
+      };
+    });
+  };
+
+  const getDataSourceOptions = () => {
+    const { alerting, tracing, metrics, mixed, dashboard, variables, annotations, pluginId, type, filter, logs } =
+      props;
+
+    const options = dataSourceSrv
+      .getList({
+        alerting,
+        tracing,
+        metrics,
+        logs,
+        dashboard,
+        mixed,
+        variables,
+        annotations,
+        pluginId,
+        filter,
+        type,
+      })
+      .map((ds) => ({
+        value: ds.name,
+        label: `${ds.name}${ds.isDefault ? ' (default)' : ''}`,
+        imgUrl: ds.meta.info.logos.small,
+        meta: ds.meta,
+      }));
+
+    return options;
+  };
+
+  const {
+    autoFocus,
+    onBlur,
+    onClear,
+    openMenuOnFocus,
+    placeholder,
+    width,
+    inputId,
+    disabled = false,
+    isLoading = false,
+  } = props;
+
+  const options = getDataSourceOptions();
+  const value = getCurrentValue();
+  const isClearable = typeof onClear === 'function';
+
+  return (
+    <div data-testid={selectors.components.DataSourcePicker.container}>
+      <MultiSelect
```

Comment:
I think one feature I'd love to see is dividing the list of options in the select component in 2 distinct groups:

`Data sources that manage Alert rules` and `other data sources`. 

As a user I'd love to be able to quickly select from a set of data sources (Prometheus and Loki) that are have `Manage alerts via Alerting UI` turned on.

`react-select` does seem to support Grouped dropdowns but I'm not sure if this is exposed in our `MultiSelect` wrapper 🤔